mitosheet/mitosheet/step_performers/export_to_file.py

```diff
@@ -0,0 +1,110 @@
+
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+
+from time import perf_counter
+from typing import Any, Dict, List, Optional, Set, Tuple
+from mitosheet.code_chunks.code_chunk import CodeChunk
+from mitosheet.code_chunks.export_to_file_code_chunk import ExportToFileCodeChunk
+from mitosheet.excel_utils import get_df_name_as_valid_sheet_name
+import pandas as pd
+import os
+
+from mitosheet.state import State
+from mitosheet.step_performers.step_performer import StepPerformer
+from mitosheet.step_performers.utils import get_param
+
+def get_export_to_csv_sheet_index_to_file_name(state: State, file_name: str, sheet_indexes: List[int]) -> Dict[int, str]:
+    if len(sheet_indexes) == 1:
+        return {sheet_indexes[0]: f'{file_name}'}
+    else:
+        file_name_without_extension, extension = os.path.splitext(file_name)
+        return {sheet_index: f'{file_name_without_extension}_{state.df_names[sheet_index]}{extension}' for sheet_index in sheet_indexes}
+
+def get_export_to_excel_sheet_index_to_sheet_name(state: State, file_name: str, sheet_indexes: List[int]) -> Dict[int, str]:
+    return {sheet_index: get_df_name_as_valid_sheet_name(state.df_names[sheet_index]) for sheet_index in sheet_indexes}
+
+
+def get_final_file_name(file_name: str, _type: str) -> str:
+    _, extension = os.path.splitext(file_name)
+    if extension == '':
+        if _type == 'csv':
+            return file_name + '.txt'
```

Comment:
Why `.txt`  and not `.csv`?

Reply:
We could give the user another .txt option if you think that's important. But I think the .csv option should generate a csv

Reply:
A .csv is a .txt file. But we can switch, nbd.